Logbook.org

```diff
@@ -0,0 +1,293 @@
+
+#+begin_example shell
+  cd queries
+  for q in ??-*.rq; do ./timer.sh $query http://128.176.233.7:3030/OME/sparql ontop 30
+#+end_example
+
+*** Results
+:PROPERTIES:
+:ID:       79eb7d26-fc85-4802-936a-b9a9bccf2f62
+:END:
+[[file:queries/saved_runs/20241210T091800/facet_walltime.png]]
+
+*** Discussion
+:PROPERTIES:
+:ID:       9fe63540-6427-44bc-8e10-07d53b0e0805
+:END:
+In all queries, ontop's query response time is faster than fuseki. Must be aware of caching, hence should try other query submission strategy (-> see Run 2, Run 3)
+
+** Run 2: Loop over 10 queries, 2 servers, 10 rounds in that order
+:PROPERTIES:
+:ID:       825c8799-bb59-4b89-aa22-1b4b2d1279c8
+:END:
+By changing the order with respect to Run1, we try to eliminate the effect of caching, as each server never sees the same two queries immediately one after the other, wherea
+before, all 30 queries were run one after the other.
+:PROPERTIES:
+:ID:       83d34711-801f-4a7e-a407-772b00ca12b0
+:END:
+    :PROPERTIES:
+    :CUSTOM_ID: run-2-run-all-10-queries-one-after-the-other-first-on-fuseki-then-on-ontop.-repeat-30-times.
+    :END:
+*** Run parameters
+     :PROPERTIES:
+     :CUSTOM_ID: run-parameters
+     :ID:       f542bd95-cbc1-4eee-9c36-fbddeaf7787e
+     :END:
+- Servers: ontop, fuseki
+- number of query repeats: 30
+- client ip: 128.176.254.12
+- client hostname: micropop046
+- client software: apache-jena-5.2.0/bin/rsparql
+- rsparql version reported as 4.2.0
+- connection: vpn to uni-muenster
+- note: changing order of looping over endpoints, rounds, and queries,
+  see script below.
+- data generating script: =run_all_queries.sh=
+
+*** Script
+     :PROPERTIES:
+     :CUSTOM_ID: script
+     :ID:       15971162-b06e-4025-a33b-6ac3ac6c8b88
+     :END:
+The script was run while connected to the VPN. Adjust the IP if you take
+this as basis for your own runs.
+
+#+begin_example shell
+#! /bin/bash
+
+for query in ??-*.rq; do
+    echo "Wall (s),User (s),Sys (s)" > $(basename ${query} .rq).ontop.timings.collected.csv
+    echo "Wall (s),User (s),Sys (s)" > $(basename ${query} .rq).fuseki.timings.collected.csv
+done
+
+for round in {1..30}; do
+    echo "round=$round "
+    for query in ??-*.rq; do
+        echo "$query "
+
+        echo ontop
+        ./timer.sh $query http://10.14.28.137:8080/sparql ontop 1
+        cat $(basename ${query} .rq).ontop.timings.csv | tail -1 >> $(basename ${query} .rq).ontop.timings.collected.csv
+
+        echo fuseki
+        ./timer.sh $query http://10.14.28.137:3030/OME/sparql fuseki 1
+        cat $(basename ${query} .rq).fuseki.timings.csv | tail -1 >> $(basename ${query} .rq).fuseki.timings.collected.csv
+    done
+done
+#+end_example
+
+*** Results
+     :PROPERTIES:
+     :CUSTOM_ID: results-1
+     :ID:       e49db261-4cf9-4b44-823c-6045b3984f17
+     :END:
+Timings come out differently: for many queries, now fuseki is faster,
+ontop factor 2 slower.
+[[file:queries/saved_runs/20241210T113800/facet_walltime.png]]
```

Comment:
Any idea why?

Reply:
Well this was what we expected, right? ontop has the additional task to rewrite a received sparql query into sql and re-translate the sql response into rdf.